sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/JdbcIOTest.java

```diff
@@ -105,193 +103,39 @@ public void setup() throws Exception {
-                        payload = resultSet.getString(i);
-                        break;
-                      case Types.NCHAR:
-                        payload = resultSet.getNString(i);
-                        break;
-                      case Types.NCLOB:
-                        payload = resultSet.getNClob(i);
-                        break;
-                      case Types.SMALLINT:
-                        payload = resultSet.getInt(i);
-                        break;
-                      case Types.TIME:
-                        payload = resultSet.getTime(i);
-                        break;
-                      case Types.TIMESTAMP:
-                        payload = resultSet.getTimestamp(i);
-                        break;
-                      case Types.TINYINT:
-                        payload = resultSet.getInt(i);
-                        break;
-                      case Types.VARCHAR:
-                        payload = resultSet.getString(i);
-                        break;
-                      default:
-                        payload = resultSet.getObject(i);
-                        break;
-                    }
-                    record.getColumnValues()[i - 1] = payload;
-                  }
+                  KV<Integer, String> kv =
+                      KV.of(resultSet.getInt("id"), resultSet.getString("name"));
+                  return kv;
                 } catch (Exception e) {
-                  LOGGER.error("Can't map row", e);
+                  return null;
                 }
-                return record;
               }
-            })).setCoder(SerializableCoder.of(JdbcDataRecord.class));
+            }))
+        .setCoder(KvCoder.of(
+            SerializableCoder.of(Integer.class), SerializableCoder.of(String.class)));
 
     PAssert.thatSingleton(
-        output.apply("Count All", Count.<JdbcDataRecord>globally()))
+        output.apply("Count All", Count.<KV<Integer, String>>globally()))
         .isEqualTo(1000L);
 
     PAssert.that(output
         .apply("Map Scientist", MapElements.via(
-            new SimpleFunction<JdbcDataRecord, KV<String, Void>>() {
-              public KV<String, Void> apply(JdbcDataRecord input) {
-                // find NAME column id
-                int index = -1;
-                for (int i = 0; i < input.getColumnNames().length; i++) {
-                  if (input.getColumnNames()[i].equals("NAME")) {
-                    index = i;
-                    break;
-                  }
-                }
-                if (index != -1) {
-                  return KV.of(input.getColumnValues()[index].toString(), null);
-                } else {
-                  return KV.of(null, null);
-                }
+            new SimpleFunction<KV<Integer, String>, KV<String, Void>>() {
+              public KV<String, Void> apply(KV<Integer, String> input) {
+                return KV.of(input.getValue(), null);
```

Comment:
Oh, I see, you're flipping the Integer, String into String, Void.
Instead of that, how about using Values.create() and Count.perElement() to count the String's directly?